pkgs/development/python-modules/llm-claude-3/default.nix

```diff
@@ -0,0 +1,52 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  anthropic,
+  llm,
+  pytestCheckHook,
+  pytest,
+  pytest-recording,
+  nix-update-script,
+}:
+
+buildPythonPackage rec {
+  pname = "llm-claude-3";
+  version = "0.4";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "simonw";
+    repo = "llm-claude-3";
+    rev = "refs/tags/${version}";
+    hash = "sha256-5qF5BK319PNzB4XsLdYvtyq/SxBDdHJ9IoKWEnvNRp4=";
+  };
+
+  build-system = [ setuptools ];
+  buildInputs = [ llm ];
```

Comment:
Why is it in `buildInputs` and not `dependencies` ?

Reply:
these plugins are primarily meant to be used with the `llm` CLI tool; they are bundled such that `llm.withPlugins` accepts a list of Python packages that are added to the tool's environment like so:

```nix
let pkgs = import <nixpkgs> {};
in pkgs.mkShell {
  buildInputs = [
    (pkgs.python3.pkgs.llm.withPlugins [
      pkgs.python3.pkgs.llm-claude-3
    ])
  ];
}
```

...if `llm` is added to `dependencies` then it runs the risk that multiple versions of `llm` may be present in the output closure, leading to this error:
```
Running phase: pythonCatchConflictsPhase
Found duplicated packages in closure for dependency 'llm':
  llm 0.14 (/nix/store/f4jd27y8wzpn438v4wx58djy5bsyx3hd-llm-0.14)
    dependency chain:
      this derivation: /nix/store/f4jd27y8wzpn438v4wx58djy5bsyx3hd-llm-0.14
  llm 0.14 (/nix/store/hg8vjjvzpxy8q5m557b819vaasqlg8l1-python3.11-llm-0.14)
    dependency chain:
      this derivation: /nix/store/f4jd27y8wzpn438v4wx58djy5bsyx3hd-llm-0.14
      ...depending on: /nix/store/dmn2f1vk4a3ba30v7jclhf6980q9zqdx-python3.11-llm-claude-3-0.4
      ...depending on: /nix/store/hg8vjjvzpxy8q5m557b819vaasqlg8l1-python3.11-llm-0.14

Package duplicates found in closure, see above. Usually this happens if two packages depend on different version of the same dependency.
```

`llm-claude-3` will only ever be used with `llm`, so there's no risk that this dependency will fail to be propagated; adding it to `buildInputs` means that we can actually build the package and use it.

Reply:
i suppose this would also work if we added it to `build-system`, although that feels less clear than `buildInputs` even though this isn't a link-time dependency